internal/helmdeployer/lookup.go

```diff
@@ -0,0 +1,87 @@
+package helmdeployer
+
+import (
+	"text/template"
+	"text/template/parse"
+
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+// hasLookupFunction checks if any template in the given Helm chart
+// calls the "lookup" function. It parses the templates to ensure it's a function
+// call and not just the word "lookup" in text or comments.
+func hasLookupFunction(ch *chart.Chart) bool {
+	for _, tpl := range ch.Templates {
+		// Parse the template into an AST.
+		t, err := template.New(
+			tpl.Name,
+		).Option(
+			"missingkey=zero",
+		).Funcs(
+			map[string]interface{}{"lookup": func() error { return nil }},
+		).Parse(string(tpl.Data))
+		if err != nil {
+			// Some templates might not parse correctly if they depend on values
+			// that aren't available. We can safely ignore these errors and continue,
+			// as a parse error means we couldn't definitively find a valid 'lookup' call.
+			continue
+		}
+
+		// Walk all parse trees in this template and look for lookup invocations.
+		if t.Tree != nil && t.Root != nil {
+			if containsLookup(t.Root) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// containsLookup recursively checks whether a parse.Node (and its children)
+// contains a call to the "lookup" function.
+func containsLookup(node parse.Node) bool {
+	if node == nil {
+		return false
+	}
+
+	switch n := node.(type) {
+	case *parse.ActionNode:
+		return containsLookup(n.Pipe)
+	case *parse.IfNode:
+		return containsLookup(n.Pipe) || containsLookup(n.List) || containsLookup(n.ElseList)
+	case *parse.ListNode:
+		for _, subNode := range n.Nodes {
```

Comment:
nit: according to my editor, this could be "simplified" using [slices.ContainsFunc](https://pkg.go.dev/slices#ContainsFunc). No strong opinion here though, since saving one line of code does not result in better readability in my opinion.

Same remark about L79 (and I'd add L66, which wasn't flagged).